client/src/hash.js

```diff
@@ -1,14 +1,22 @@
 // Get the hash of the url
-const hash = window.location.hash
-  .substring(1)
-  .split("&")
-  .reduce(function(initial, item) {
-    if (item) {
-      var parts = item.split("=");
-      initial[parts[0]] = decodeURIComponent(parts[1]);
-    }
-    return initial;
-  }, {});
-window.location.hash = "";
+export const authorize = () => {
+  var location = window.location;
+  const search = location.search.substring(1).split("=");
+  window.history.replaceState({}, document.title, "/home")
+  return search && search[0] === "code" ? search[1] : "";
+};
 
-export default hash;
+export const hash = () => {
+  const h = window.location.hash
```

Comment:
I know you use "export default hash" but I do believe we should change this from "const h" to a different variable name than just "h".